vlo-web-app/src/main/java/eu/clarin/cmdi/vlo/wicket/provider/SolrDocumentExpansionPairProvider.java

```diff
@@ -22,36 +22,84 @@
 import java.util.Iterator;
+import java.util.List;
+
 import org.apache.wicket.markup.repeater.data.IDataProvider;
 import org.apache.wicket.model.IModel;
 
+import eu.clarin.cmdi.vlo.VloWebSession;
+import org.apache.wicket.protocol.http.request.WebClientInfo;
+import org.apache.wicket.request.cycle.RequestCycle;
+import org.slf4j.LoggerFactory;
+
+import com.sun.istack.logging.Logger;
+
+import org.apache.wicket.protocol.http.servlet.ServletWebRequest;
+import eu.clarin.cmdi.vlo.pojo.QueryFacetsSelection;
+import eu.clarin.cmdi.vlo.exposure.models.SearchQuery;
+import eu.clarin.cmdi.vlo.exposure.models.SearchResult;
+import eu.clarin.cmdi.vlo.config.ServletVloConfigFactory;
+import eu.clarin.cmdi.vlo.config.VloConfig;
+
 /**
  *
  * @author twagoo
  */
 public class SolrDocumentExpansionPairProvider implements IDataProvider<SolrDocumentExpansionPair> {
+    private final static org.slf4j.Logger logger = LoggerFactory.getLogger(SolrDocumentExpansionPairProvider.class);
 
     private final FieldNameService fieldNameService;
 
     private final IModel<QueryFacetsSelection> selectionModel;
 
     private Long size;
 
-    public SolrDocumentExpansionPairProvider(IModel<QueryFacetsSelection> selection, FieldNameService fieldNameService) {
+    public SolrDocumentExpansionPairProvider(IModel<QueryFacetsSelection> selection,
+            FieldNameService fieldNameService) {
         this.selectionModel = selection;
         this.fieldNameService = fieldNameService;
     }
 
     @Override
     public Iterator<? extends SolrDocumentExpansionPair> iterator(long first, long count) {
-        final SolrDocumentExpansionList documents = getDocumentService().getDocumentsWithExpansion(selectionModel.getObject(),
-                BigDecimal.valueOf(first).intValueExact(), // safe long->int conversion
-                BigDecimal.valueOf(count).intValueExact(),
-                FacetConstants.COLLAPSE_FIELD_NAME); // safe long->int conversion
+        final SolrDocumentExpansionList documents = getDocumentService().getDocumentsWithExpansion(
+                selectionModel.getObject(), BigDecimal.valueOf(first).intValueExact(), // safe long->int conversion
+                BigDecimal.valueOf(count).intValueExact(), FacetConstants.COLLAPSE_FIELD_NAME); // safe long->int
+                                                                                                // conversion
+
+        VloConfig vloConfig = VloWicketApplication.get().getVloConfig();
+        if (vloConfig.isVloExposureEnabled()) {
+            try {
+                // these values to be saved in postgresql to calculate record-exposure
+                // get page url
+                String pageUrl = ((ServletWebRequest) RequestCycle.get().getRequest()).getContainerRequest()
+                        .getRequestURL().toString();
+                // get user ip address
+                String ip = ((WebClientInfo) VloWebSession.get().getClientInfo()).getProperties().getRemoteAddress();
```

Comment:
I believe we have a policy of not storing IP address as they are considered personal information. We can store them 'anonymised'. But I'm not sure if we need to store them at all. Did you have a specific reason or use case in mind?